service/dap/server.go

```diff
@@ -516,28 +516,35 @@ func (s *Session) address() string {
 func (s *Session) ServeDAPCodec() {
 	// Close conn, but not the debugger in case we are in AcceptMulti mode.
 	// If not, debugger will be shut down in Stop().
+	var (
+		err     error
+		request dap.Message
+	)
 	defer s.conn.Close()
+	defer func() {
```

Comment:
I don't think this refactoring is actually a refactoring. The original code only does this after checking `s.config.StopTriggered`.

Reply:
You're right, we should add a sentinel value to determine if we should call this, good catch. I just wanted to avoid a `defer` call within the loop, especially because that loop has a `continue` which could cause a leak.

Reply:
Done.